### ISSUE TITLE
Request: STAR for small DNA fragment alignment to diploid genome (specific reason why not BWA)

### ISSUE THREAD

**Issue by kingralph80** [alexdobin/STAR#968]
Hi,
I know there have been threads about STAR and DNA alignment. However, I have a few specific questions and reason why STAR and not BWA. We are trying to map small(65 bp) fragment coming from an F1. For allele-specific analysis personalized genomes left to much mapping bias, so we use a diploid genome of both parents. This works great for unique reads but 75% of reads map ambiguous in diploid alignment, due to no variation between the parent genomes. We loose a lot of TF peaks if we remove them. 

The key is that STAR has an option to limit multi-map reads in the output. We want to keep multi-map reads that map exactly twice. BWA only offers MAPQ=0. 
Question:
1) STAR determines how many times a read with the same score maps? So we can set to cutoff to two reads with --outSAMmultNmax 2?
2) What MAPQ do multi-map reads with 100% equal mapping to two places get?
3) Are there any parameters I should change to map single DNA fragments (~65 bp) to a reference with STAR?

**Comment by alexdobin** [comment]
Hi @kingralph80 

1. --outSAMmultNmax 2 will only keep reads that map to 1 or 2 loci, and discard all others. If you are not interested "true" multimappers, you can set this threshold. Note that you will still get some reads that map to 2 unrelated loci, so you will need a way to filter those out.

2. MAPQ output is presently calculated from the number of loci the read maps to, as 10*log10(1-1/Nloci)=3 for Nloci=2

3. For DNA mapping you would want to prohibit splicing with --alignIntronMax 1 . Other parameters should be fine, I believe.

Cheers
Alex

**Comment by thomashartwig80** [comment]
Hi Alex thanks for the feedback. 

I thought about the issue a bit more. Using --outSAMmultNmax 2 should allow three possibilities: Case 1) A read maps once to each haplotype in the diploid (those I want), Case 2) A Read maps twice to one haplotype but not the other haplotype (those are tricky), or Case 3) a read maps twice to two unrelated loci e.g. because of a sequencing error or assembly error or simple bad mappability. 

Case 3: I don't think this one can be dealt with. As you mentioned in the email, with SNPs there are ways to collapse reads to a reference, but we heavily rely on indels. We have a chain file but WGA is not good enough for such a collapse yet. 

Case 2: This one I thought about and I actually think its better to keep those reads than to remove them. Scenario you have an identical duplication in haplotype 1 and 1 copy of the loci but with a SNP in haplotype 2. If we remove the reads that multi-map to haplotype 1 it creates an artefact with the reads that map to haplotype 2. So I think its saver to keep reads from Case2 and distribute them randomly.

**Comment by alexdobin** [comment]
Hi Thomas,

good summary of the issues with diploid alignments.
The best approach to understand whether the 2-time mappers are truly from the same locus in two haplotypes is to convert both alignments back to the reference - but that indeed requires good assemblies for the haplotypes.

You might be able to deal with this issue in the downstream processing - by processing the haplotypes separately. For instance, if you are counting reads per gene, you can have gene annotations for each haplotype, and the reads that map to the same locus will be counted towards the same gene on both haplotypes.

Cheers
Alex

**Comment by kingralph80** [comment]
I agree that mapping them back to reference is the best way as long as they are not CNV. This is also much better for visualizing the results later.  We do actually map them back later via chain/hal files so we can count them and make ratios. 
We ran the analysis and the results look much better with the 2 cutoff STAR approach compared to our BWA results. Still debating on the pseudo-random vs full-random option but the results for peaks without a genetic difference finally look good. Lol looks like we go with STAR for DNA mapping now too. 
 
There is actually one thing that surprised me. We do get much more unique reads as well. It was not clear to me from the manual at which point STAR gives a reads 255 as unique.

**Comment by alexdobin** [comment]
Sounds good! Once you get this pipeline published, please post a link!
The MAPQ=255 is arbitrary for unique mappers, it can be set to any value with `--outSAMmapqUnique`
The real cutoff for multimappers is defined by `--outFilterMultimapScoreRange`, which sets the range of scores from the top score for multimappers to be reported. By default it's =1, which means that all alignments with score (topScore-1) will be considered multimappers and given MAPQ=255. One mismatch reduces the score by 2, so an alignment with one extra mismatch compared to the top alignment will not be included as multimapping.

**Comment by ColeWunderlich** [comment]
Hey Alex,

I am trying to develop an intuitive understanding of what the default `--outFilterMultimapScoreRange = 1` means.

Under the default settings, are there any single operations/penalties that could cause something to be 1 less than the topScore?  All I was able to find was `--scoreStitchSJshift` but I am not exactly sure what this signifies. 

Is it also true that one mismatch used to be a -1 penalty, which means the default `--outFilterMultimapScoreRange = 1` used to be able to be thought of as roughly things one mismatch off?

**Comment by alexdobin** [comment]
Hi Cole,

the most common case would be the alignments that are soft-clipped by one base - they will have -1 score compared to unclipped alignments. More complex cases are alignments with mismatches and indels where sum of penalties differ by -1.

**Comment by ColeWunderlich** [comment]
Thanks Alex!  The soft-clipping definitely helps provide an intuitive understanding.
